contracts/StrategyCurveTwoPool.sol

```diff
@@ -0,0 +1,383 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+
+// These are the core Yearn libraries
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/math/Math.sol";
+
+import { ICurveGauge, ICurveGaugeFactory, ICurvePool } from "../interfaces/ICurveTwoPool.sol";
+import { ISwapRouter } from "../interfaces/ISwapRouter.sol"; // Uni V3
+import { BaseStrategy, StrategyParams } from "@yearnvaults/contracts/BaseStrategy.sol";
+
+// To avoid the "stack too deep" error
+struct ContractAddress {
+  address vault;
+  address healthCheck;
+  address usdt;
+  address usdc;
+  address weth;
+  address crv;
+  address router;
+  address gauge;
+  address gaugeFactory;
+  address pool;
+}
+
+abstract contract StrategyCurveBase is BaseStrategy {
+  using Address for address;
+
+  /* ========== STATE VARIABLES ========== */
```

Comment:
no keepCRV?

Reply:
I removed this variable along with the voter, etc. as we cannot currently vote on sidechains. I'm happy to add it back if we have a reasonable expectation voting will eventually make it to arbitrum.

Reply:
I would leave keepCRV, because even if we don't necessarily vote, we probably want to keep and lock of 10% of the CRV from harvests if possible

Reply:
Absolutely. I'll get the code back.